code/game/objects/items/tools/powertool.dm

```diff
@@ -0,0 +1,86 @@
+//Power Tools
+/obj/item/powertool
+	var/stateswap = 0 //what state is it in
+	var/icon_state_swap = 0 //which icon should it be
+	var/tool_behaviour_swap = 0 //what tool behaviour(s) should be here
+	var/swapsound = 0 //the sound that plays for swapping
+
+/obj/item/powertool/attack_self(mob/living/carbon/user) //activating the item causes it to swap icons and behaviours
+	stateswap = !stateswap
```

Comment:
It would be much neater if this could support more than two states. Something like
```dm
/obj/item/powertool
  var/list/tool_behaviors= list()
  var/list/tool_icon_states = list()
  var/active_behavior_index = 1

/obj/item/powertool/attack_self(mob/living/carbon/user)
  active_behavior_index = (active_behavior_index + 1) % possible_tool_behaviors.len + 1
  tool_behaviour  = possible_tool_behaviors[active_behavior_index]
  icon_state = tool_icon_states[active_behavior_index]
...
/obj/item/powertool/drill
  ...
  tool_behaviors = list(TOOL_SCREWDRIVER, TOOL_WRENCH)
  tool_icon_states = list("drill_screw", "drill_bolt")
```

(But given that none of the tools currently *need* this, this might not be necessary)

Reply:
still good for posterity

Reply:
lets go a bit further and add on examine as well.